app/models/invite.rb

```diff
@@ -182,7 +182,9 @@ def redeem(email: nil, username: nil, name: nil, password: nil, user_custom_fiel
 
   def self.redeem_from_email(email)
     invite = Invite.find_by(email: Email.downcase(email))
-    InviteRedeemer.new(invite: invite, email: invite.email).redeem if invite
+    if invite.present? && invite.redeemable?
```

Comment:
I feel like the guard for whether an invite is reedemable or not should be done as part of `InviteRedeemer#redeem`. Otherwise, we're not centralising the logic of when an invite is supposed to be redeemable.

Reply:
I agree, will make it so. :)

Reply:
Okay, I added a check for this in `InviteRedeemer#can_redeem_invite?`.